### DISCUSSION TITLE
Init IS310 Homework

### DISCUSSION THREAD

**Discussion by ZoeLeBlanc** [CultureAsData-UIUC/is310-fall-2024#1] [Assignments]
Please add a link below to your first Init IS310 Homework [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/03-intro-versioning-git/#homework-init-is310](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/03-intro-versioning-git/#homework-init-is310).

**Comment by yosefstifanos** [comment]
Yosef Stifanos (yds2): https://github.com/yosefstifanos/is310-coding-assignments/blob/main/init-is310-homework/README.md

**Reply by ZoeLeBlanc** [reply]
Thanks for posting this Yosef! Overall your repository looks great. There's only one issue I noticed. You've added a few `.DS_Store` files which is Mac specific environment files. Best practices is to not include those, so I would encourage you to try following these instructions to remove them https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/05-advanced-git-github#what-to-do-if-you-forget-to-add-a-file-to-gitignore . We haven't covered `.gitignore` files yet, so you can wait until we do on Tuesday. Otherwise great work 👍🏽

**Reply by yosefstifanos** [reply]
I didn't know what a .DS_Store file was. I've seen it a lot, but just never knew. Should be fixed now. Can you check to see if I did it correctly? @ZoeLeBlanc

**Reply by yosefstifanos** [reply]
https://github.com/yosefstifanos/is310-coding-assignments/

**Reply by ZoeLeBlanc** [reply]
That looks great! One small note that you do not need to do know but eventually will be helpful is that you `.gitignore` should be in the top level folder of your project folder (in this case `is310-coding-assignments`). Otherwise, it may not apply to other folders you for homework assignments. Again you don't need to do it today, but would recommend doing it eventually. Otherwise you will need to make one for each homework assignment. Great job on fixing this so quickly though 👍🏽

**Reply by yosefstifanos** [reply]
@ZoeLeBlanc Should be fixed. Can you check? I also had to re-remove the `.DS_Store` file again. Idk why it returned back again.

**Reply by ZoeLeBlanc** [reply]
Looks good to me 👍🏽 !

**Comment by esr27** [comment]
https://github.com/esr27/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi @esr27 ! Your homework response looks great but there's few issues, you need to resolve. First, I don't actually know who this is, so could you include your name here. Second, it looks like you posted the `init-is310-homework` folder as your main folder. Instead you should have all these materials within this folder, and that folder should be in your `is310-coding-assignments` folder. So you should have the following structure:
```
is310-coding-assignments/
└── init-is310-homework/
    ├── images/
    │   └── image1.jpg
    └── README.md
```
Finally, you also pushed up some files that we usually don't include in a GitHub repository. Specifically `.DS_Store` files, which are Mac specific environment files. We want to exclude these because they can cause a lot of conflict in version histories. You can follow these instructions [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/05-advanced-git-github#what-to-do-if-you-forget-to-add-a-file-to-gitignore](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/05-advanced-git-github#what-to-do-if-you-forget-to-add-a-file-to-gitignore) to remove the file and/or wait for our class Tuesday where will be covering this materials. Let me know if you have any questions and otherwise, great work 👍🏽

**Reply by esr27** [reply]
Thanks for the advice. I fixed the folders. This is Ethan Roberts, by the way.

**Comment by AlanK3** [comment]
https://github.com/AlanK3/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi Alan! This largely looks great but it looks like you accidentally made the `README.md` for `init-310-homework` the main `README.md` for the entire repository. If you could move the current `README.md` into the correct folder that would be great, and we will next week be adding another `README.md` for the entire `is310-coding-assignments`. Otherwise, this is great work 👍🏽

**Reply by AlanK3** [reply]
Thanks for the heads up! Should be fixed now.

**Reply by ZoeLeBlanc** [reply]
Hi @AlanK3! Just a small update, unfortunately your images are no longer rendering (I get the following message `This content is no longer available.`). Would you mind fixing that? Thanks!

**Comment by haydnelm** [comment]
https://github.com/haydnelm/is310-coding-assignments

**Comment by ryanpearlman1** [comment]
https://github.com/ryanpearlman1/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi Ryan! Thanks for sharing this and have a few tweaks for you. First, for this assignment, you are missing your **Discord** and **Hypothesis** handles from your `README.md`, so please add those in to receive full marks.

Second, it looks like you've made your command line maze assignment the main `README.md` file for your `is310-coding-assignments`. However, the assignment should go into a separate folder within the `is310-coding-assignments` called `command-line-maze`. So you should have the following directory structure:

```is310-coding-assignments/
├── .gitignore
├── README.md
├── init-is310-homework/
│   ├── images/
│   └── README.md
└── command-line-maze/
    ├── command_line.zip
    └── README.md
```

Happy to help get the directory sorted and you don't need to fix it for tomorrow but you will need this fixed for Thursday, so please don't hesitate to reach out if you have questions. Good work so far 👍🏽

**Comment by GloriaLiu930** [comment]
https://github.com/GloriaLiu930/is310-coding-assignments

**Comment by averyWebster** [comment]
https://github.com/averyWebster/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi Avery! Overall this looks great but a few tweaks for your submission of the command line maze. Currently you have the maze in your main directory rather than in the correct folder. You also have a `.DS_Store` file pushed up to GitHub.

So I would recommend, that you move your zip folder into the `command-line-maze` folder and also create a `.gitignore` file. This is a special type of file that tells git what we **do not** want to push up to our remote GitHub repository. You can read more about the file here [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/05-advanced-git-github#gitignore](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/05-advanced-git-github#gitignore), but all you need to do is follow the steps:

1. Create a `.gitignore` locally in your `is310-coding-assignments` folder
2. Add `.DS_Store` to that file and save it.
3. Delete your local `.DS_Store`.
4. Type `git rm --cached .DS_Store` in your terminal
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Happy to help with this if you have questions, but again overall good work 👍🏽

**Comment by OctaneCodes** [comment]
Nick Cook
https://github.com/OctaneCodes/is310-coding-assignments/tree/main/init-is310-homework

**Reply by ZoeLeBlanc** [reply]
Hi Nick! This looks great but I did notice a bit of an issue with your folder names. Specifically your GitHub repo is currently named `IS310Repo` but it should be `is310-coding-assignments`. You can change this in your settings for the repository. That will obviously mean you will want to change your local setup as well.

This might be a bit tricky so happy to help you after class or setup a Zoom call. Essentially, you want your `git` to live in your `is310-coding-assignments` locally rather than `is310Repo`. So what I would do is rename your local `IS310Repo` (or whatever the folder is called locally) to `is310-coding-assignments`. Then I would move init-is310-homework out of the nested `is310-coding-assignments` and then delete that nested folder. You should be able to then have the following structure locally:

```is310-coding-assignments/
├── .gitignore
├── README.md
├── init-is310-homework/
│   ├── images/
│   └── README.md
```

You can then use our normal git workflow of `git add .`, `git commit -m "fixed directory structure`, and then `git push origin main` though do reach out if you have any issues. Good work so far though 👍🏽

**Reply by OctaneCodes** [reply]
Should be fixed. Was really easy going at it locally

**Reply by ZoeLeBlanc** [reply]
Looks perfect! Just updated your link above to match the new structure but great work 👍🏽

**Comment by tinayang922** [comment]
Weiting Yang
https://github.com/tinayang922/is310-coding-assignments/tree/main/init-is310-homework

**Reply by ZoeLeBlanc** [reply]
Hi Weiting! Overall this looks great, but I did notice you have a `.DS_Store` file in each of your folders. We try to avoid adding those files to our remote GitHub for a few reasons (clutters a repo and causes issues if someone tries to run your code on a different computer). You an easily remove them with the following steps:

1. Create a `.gitignore` locally in your `is310-coding-assignments` folder
2. Add `.DS_Store` to that file and save it.
3. Delete your local `.DS_Store` in each folder.
4. Type `git rm --cached .DS_Store` in your terminal.
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Let me know if you have any issues following these instructions but otherwise everything looks great 👍🏽

**Reply by tinayang922** [reply]
Thank you for the feedback and assistance in fixing this. I pushed the latest version without .DS_Store.

**Reply by ZoeLeBlanc** [reply]
Looks perfect! Well done 👍🏽

**Comment by Cindy1226** [comment]
Cindy Liang
https://github.com/Cindy1226/is310-coding-assignments/tree/main/init-is310-homework

**Reply by ZoeLeBlanc** [reply]
Hi Cindy! Overall this looks great, but I did notice you have a `.DS_Store` file in each of your folders. We try to avoid adding those files to our remote GitHub for a few reasons (clutters a repo and causes issues if someone tries to run your code on a different computer). You an easily remove them with the following steps:

1. Create a `.gitignore` locally in your `is310-coding-assignments` folder
2. Add `.DS_Store` to that file and save it.
3. Delete your local `.DS_Store` in each folder.
4. Type `git rm --cached .DS_Store` in your terminal.
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Let me know if you have any issues following these instructions but otherwise everything looks great 👍🏽

**Comment by ahmadsawwan** [comment]
Ahmad Sawwan
https://github.com/ahmadsawwan/ahmadsawwan-is310-coding-assignments/tree/main

**Reply by ZoeLeBlanc** [reply]
Hi Ahmad! Overall this looks great but I noticed you had two `.gitignore` files and that you also have `.DS_Store` files that we usually try to not push up to GitHub. You can fix this by following these steps.

1. Delete the `.gitignore` in your `init-is310-homework` folder. We only need one `.gitignore` per `.git` (so only when you do `git init` should you be creating a `.gitingore` and usually at the same folder level, if that makes sense).
2. Add `.DS_Store` to your existing `.gitignore` file and save it.
3. Delete your local `.DS_Store` in each folder.
4. Type `git rm --cached .DS_Store` in your terminal.
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Let me know if you have any issues following these instructions but otherwise everything looks great 👍🏽

**Reply by ahmadsawwan** [reply]
Hello! I think I just fixed it, can you check and let me know if I did everything correctly?

**Reply by ZoeLeBlanc** [reply]
Looks perfect! Well done 👍🏽

**Comment by adg77** [comment]
https://github.com/adg77/is310-coding-assignments.git
Alex Guzman

**Reply by ZoeLeBlanc** [reply]
Hi Alex! This looks great except one small issue. You've shown proof that you have GitHub but not `git` installed on your computer. If you could add a screenshot of the output of `git --version` in your spot for git then you'll get full marks 👍🏽 .

Also I noticed you deleted your `.DS_Store` file which is great. To prevent you from pushing it up accidentally, I would highly recommend making a `.gitignore` file with the following steps:

1. Create a `.gitignore` file in your `is310-coding-assignments` folder. You only need one per `.git` folder.
2. Add `.DS_Store` to your `.gitignore` file and save it.
5. Do your normal `git add .`, `git commit -m "Created a .gitignore file", and finally `git push origin main`. 

Happy to answer any questions but otherwise good work so far 👍🏽

**Comment by divyaj5** [comment]
Divya Jain (divyaj5): https://github.com/divyaj5/is310-coding-assignments/tree/master

**Reply by ZoeLeBlanc** [reply]
Hi Divya! This is a great start but noticed some issues you need to fix.

First, you currently have the content of the `README.md` for your `init-IS310-homework` in your main `is310-coding-assignments` folder's `README.md` [https://github.com/divyaj5/is310-coding-assignments/blob/master/README.md](https://github.com/divyaj5/is310-coding-assignments/blob/master/README.md).  You should move that content to the empty `README.md` file you have in your `init-IS310-homework` and then just have your `is310-coding-assignments` have either an empty `README.md` or add something like `# IS310 Coding Assignments` to it. Once you make this change, your images should render correctly in GitHub.

Second, I noticed you have a `.DS_Store` file in each of your folders. We try to avoid adding those files to our remote GitHub for a few reasons (clutters a repo and causes issues if someone tries to run your code on a different computer). You an easily remove them with the following steps:

1. Create a `.gitignore` locally in your `is310-coding-assignments` folder
2. Add `.DS_Store` to the top of that file and save it.
3. Delete your local `.DS_Store` in each folder.
4. Type `git rm --cached .DS_Store` in your terminal.
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Let me know if you have any issues following these instructions but otherwise think you're very close to be complete 👍🏽

**Reply by divyaj5** [reply]
Hello! Just completed those two steps. Let me know if I was able to get that fixed.

**Reply by ZoeLeBlanc** [reply]
Looks perfect! Great work 👍🏽

**Comment by officalcharles** [comment]
Havyarimana Charles 
https://github.com/officalcharles/is310-coding-assignments/tree/master/init-is310-homework

**Reply by ZoeLeBlanc** [reply]
Hi Charles! The link you posted goes to a 404 page on GitHub and when I click on your GitHub profile, I don't see the repository. Did you accidentally delete it? Please feel free to reach out on Discord if you're having issues.

**Reply by officalcharles** [reply]
I think I fixed the issue. Could you try again?

**Reply by ZoeLeBlanc** [reply]
Looks great! One of your images isn't rendering but I can see it in the folder so not a problem Otherwise great work 👍🏽

**Comment by theajk2** [comment]
Thea Kidwell (edited)
https://github.com/theajk2/is310-coding-assignments/tree/main/init-is310-homework

**Reply by ZoeLeBlanc** [reply]
Hi Thea! Overall this looks good, but some small tweaks.

First, the link you posted is to an actual commit rather than the correct folder. So please update your answer above to be this link [https://github.com/theajk2/is310-coding-assignments/tree/main/init-is310-homework](https://github.com/theajk2/is310-coding-assignments/tree/main/init-is310-homework).

Second, your `README.md` in `init-is310-homework` does not currently have any content in it. If you follow the instructions from the assignment [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/03-intro-versioning-git/#homework-init-is310](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/03-intro-versioning-git/#homework-init-is310), you should add in proof of the required course software and then your Discord and Hypothesis handles. Feel free to reach out if you have questions but you're very close 👍🏽

**Reply by theajk2** [reply]
Hello Professor!  I have made both changes suggested including moving proper documentation to README. Let me know if that didn't work!

**Comment by frangara** [comment]
Francis Angara
https://github.com/frangara/is310-coding-assignments/tree/main

**Reply by ZoeLeBlanc** [reply]
Hi Francis! This is a good start but you have a few issues with what you submitted. First, it looks like you've put your `init-is310-homework` folder `README.md` file as the main file for your `is310-coding-assignments` folder. However, that `README.md` file should be in your `init-is310-homework` folder which is why none of your image paths are rendering correctly. Your file directory should look like the following:

```├── .gitignore
├── README.md
├── init-is310-homework/
│   ├── images/
│   └── README.md
```
If you move your current `README.md` into that folder you should be good to go. 

To move the file, you can either drag and drop it with a file GUI or in VS Code. Or you can use the following commands depending on your computer.

For Macs, you need to do `mv README.md init-is310-homework/` from your `is310-coding-assignments` folder.

For PowerShell on a Windows machine, you would do `Move-Item README.md .\init-is310-homework\` again from your `is310-coding-assignments` folder.

Happy to help after class today, and overall this is a good start 👍🏽

**Reply by frangara** [reply]
Got it! Thank you! I have moved the readme file but it is still giving me a 404 error with it.

**Reply by ZoeLeBlanc** [reply]
Hi @frangara ! It looks like your image paths are slightly off `![VS Code Installation](is310-coding-assignments/init-is310-homework/images/vscode-installation.png)`. It should instead look like this `![VS Code Installation](images/vscode-installation.png)` and that should fix your 404 errors. Please let me know if you have questions still but you're very close to being complete 👍

**Comment by lhayedimarucut** [comment]
Lhaye Dimarucut https://github.com/lhayedimarucut/is310-coding-assignments

**Comment by SamuelSokolovsky** [comment]
Samuel Sokolovsky https://github.com/SamuelSokolovsky/is310-coding-assignments/tree/main

**Comment by chuzesta** [comment]
Jeff Fang https://github.com/chuzesta/is310-coding-assignments/tree/main/init-is310-homework

**Reply by ZoeLeBlanc** [reply]
Hi Jeff! Overall this looks great, but I did notice you have a `.DS_Store` file in your repository. We try to avoid adding those files to our remote GitHub for a few reasons (clutters a repo and causes issues if someone tries to run your code on a different computer). You an easily remove them with the following steps:

1. Create a `.gitignore` locally in your `is310-coding-assignments` folder
2. Add `.DS_Store` to that file and save it.
3. Delete your local `.DS_Store` file. You can do that with the following command, though be sure to run this in your `is310-coding-assignments` folder. The command would be `rm -rf .DS_Store`.
4. Type `git rm --cached .DS_Store` in your terminal. It's ok if it shows an error, we just want to remove the file from your git history.
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Let me know if you have any issues following these instructions but otherwise everything looks great 👍🏽

**Comment by LucasWashor** [comment]
Lucas Washor https://github.com/LucasWashor/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi Lucas! This looks good overall, except that it seems your missing the `init-is310-homework` folder and that your materials for that folder are now currently in the main `is310-coding-assignments`. This is very understandable but will quickly become a problem as we keep doing assignments. 

So I would recommend you create an `init-is310-homework` folder and move your `images` folder and current `README.md` file to that folder. Then you can follow are normal git workflow of: 

- `git add .` (to add all the files you've moved)
- `git commit -m "created folder and moved files"`
- `git push origin main`

After that you should be set for the assignment, so good work 👍🏽

**Comment by nchat3** [comment]
Diya Chaturvedula https://github.com/nchat3/is310-coding-assignments.git)

**Reply by ZoeLeBlanc** [reply]
Hi Diya! You're very close here but it looks like your images are not rendering correctly in your `README.md` in your `init-is310-homework` folder. I would recommend potentially renaming your images to things like `git-version.png` or `python-version.png` and then correcting the file paths in your `README.md` to be `![git version](images/git-version.png)`. The issue you're having is likely caused by either changing file names or the spaces in your image file names. Happy to help if you have questions, but overall good work 👍🏽

**Comment by Zifanzhao0801** [comment]
Zifan Zhao https://github.com/Zifanzhao0801/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi Zifan! Thanks for posting this and you're fairly close but I noticed a few issues.

First, your links to images do not render or show any file in GitHub. It seems like maybe you forgot or some didn't push up your `images` folder in your `init-is310-homework` folder. So if you do that those should render correctly.

Second, you didn't include your Discord or Hypothesis usernames in your `README.md` in your `init-is310-homework` folder. Have you also filled out the Initial Interest Survey? It was due last week but I don't see a response from you, so if you could do that as well that would be great!

Third and final, is that I noticed you have a `.DS_Store` file in your repository. We try to avoid adding those files to our remote GitHub for a few reasons (clutters a repo and causes issues if someone tries to run your code on a different computer). You an easily remove them with the following steps:

1. Create a `.gitignore` locally in your `is310-coding-assignments` folder
2. Add `.DS_Store` to that file and save it.
3. Delete your local `.DS_Store` file. You can do that with the following command, though be sure to run this in your `is310-coding-assignments` folder. The command would be `rm -rf .DS_Store`.
4. Type `git rm --cached .DS_Store` in your terminal. It's ok if it shows an error, we just want to remove the file from your git history.
5. Do your normal `git add .`, `git commit -m "deleted .DS_Store and added .gitignore", and finally `git push origin main`. 

Let me know if you have any issues following these instructions and happy to help if you have questions 👍🏽

**Comment by AG-Samaniego** [comment]
AG Samaniego
https://github.com/AG-Samaniego/is310-coding-assignments/tree/main

**Comment by RebecaRodr** [comment]
Rebeca Rodriguez
https://github.com/RebecaRodr/is310_coding_assignments/tree/main

**Reply by ZoeLeBlanc** [reply]
Hi Rebeca! This is a good start but I noticed a few issues. I'm happy to help you after class today to fix these if that would be helpful.

First, you do not have an `init-is310-homework` folder. We created folders for each of the assignments in this course in our `is310-coding-assignments` parent folder. So you should have a structure that looks like this:

```
is310-coding-assignments/
├── README.md
├── init-is310-homework/
│   ├── images/
│   └── README.md
```
Relatedly, you do not have an `images` folder but instead it looks like you create an empty file called `Images`. It can be confusing what a file is versus a folder, but you usually can click on a folder and see if there is anything inside whereas a file you cannot. You can create a folder with the following commands.

For **Windows Powershell and from your `is310-coding-assignments` folder**, you could do `ni -ItemType dir init-is310-homework` to create the first folder. Then you would enter the folder with `chdir init-is310-homework`. Finally, you would type the same command to create a folder but this time for images, so `ni -ItemType dir images`. 

For **Mac in your terminal and from your `is310-coding-assignments` folder**, you could do `mkdir init-is310-homework` to create the first folder. Then you would enter the folder with `cd init-is310-homework`. Finally, you would type the same command to create a folder but this time for images, so `mkdir images`. 

Once you have the folders created, you can move your current `README.md` file into the `init-is310-homework` folder and start adding screenshots of your installations. To do that, you would add the images to your `images` folder. You can either leave the screen shot as it is named or give it a new name like `git-version.png` or `python-version.png`. 

Then in your `README.md` file in your `init-is310-homework` folder, you add the following to make the images render correctly:

```Markdown
![Git Version](images/git-version.png)
```

After you have everything restructured and correctly added, then you just need to follow our standard git workflow of:

- `git add .` (to add all the files you've moved)
- `git commit -m "created folders, added images to README"`
- `git push origin main`

Again if you have any questions, don't hesitate to reach out and this is a great start 👍🏽

**Comment by jasonmorris8** [comment]
Jason Morris
https://github.com/jasonmorris8/is310-coding-assignments

**Comment by JBrown-0** [comment]
Justin Brown
https://github.com/JBrown-0/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi Justin! This is a great start but noticed a few issues.

First, your `README.md` file in your `init-is310-homework` folder is currently empty. I also do not see an `images` folder in that folder, so would recommend creating the images folder and then populating your `README.md` with the relevant materials. Happy to help if you have questions or get stuck!

Second, I noticed that your commits are showing up under a different username than your GitHub Handle:

![Screenshot 2024-09-10 at 10 28 49 AM](https://github.com/user-attachments/assets/4ce0f17c-c3b8-4891-af38-8822ddc15359)

This is usually a sign that your local git is not configured correctly. If you follow the steps here [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/01-course-tools/#git-configuration](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/introducing-humanities-computing/01-course-tools/#git-configuration ) of adding in your exact GitHub username and email you use to sign into GitHub with that should be fixed. Again happy to help if you have questions 👍🏽

**Comment by shoirathi** [comment]
Shoi Rathi
https://github.com/shoirathi/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi @shoirathi ! It looks like there's a problem with your is310-coding-assignments GitHub repository. Specifically you'll notice in this screenshot that some of your folders have white arrows:

<img width="1023" alt="Screenshot 2024-09-19 at 4 34 52 PM" src="https://github.com/user-attachments/assets/cf77b2c4-a9be-4e72-a03e-12da1069059e">

I believe this is because you previously did `git init` in each of these folders, creating something called `git submodules`. The big problem now is that we need to fix that locally since otherwise we can't actually click on those folders. I would be happy to meet with you to help you complete these steps or chat over Discord, so don't hesitate to reach out if that would be helpful.

Here are the steps to remove `git submodules`:

1. **Deinitialize the Submodule:**
   - Open your terminal or command line in the root directory of the repository (so in `is310-coding-assignments`.
   - Run this command to deinitialize the submodule (this breaks the submodule connection without deleting your folder):
   - The exact command will be:
     ```bash
     git submodule deinit -f command-line-maze
     ```
   - And then:
     ```bash
     git submodule deinit -f init-is310-homework
     ```
2. **Remove the Submodule from Git's Index:**
   - Run the following command to remove the submodule from the Git index, but keep the folder and its content intact:
     ```bash
     git rm --cached command-line-maze
     ```
   - And then:
     ```bash
     git rm --cached init-is310-homework
     ```
3. **Remove the Submodule Entry from `.gitmodules`:**
   - Edit the `.gitmodules` file (if it exists) and manually remove the section that refers to the submodule.
   - Look for a section that looks like this:
     ```bash
     [submodule "path/to/submodule"]
       path = path/to/submodule
       url = URL_OF_SUBMODULE
     ```
   - Remove that section, save, and close the file.

4. **Commit the Changes:**
   - After these changes, commit them to your local repository:
     ```bash
     git add .
     git commit -m "Removed submodule but kept folder"
     ```

5. **Push the Changes:**
   - Push your changes to GitHub:
     ```bash
     git push origin main
     ```

**Reply by shoirathi** [reply]
Hi I just fixed this issue, thank you for guiding me through it.

**Comment by yunchaecho2** [comment]
Yunchae Cho 
https://github.com/yunchaecho2/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi @yunchaecho2 ! I've already addressed some of the issues in your repository with this comment on a different assignment [https://github.com/CultureAsData-UIUC/is310-fall-2024/discussions/4#discussioncomment-10695817](https://github.com/CultureAsData-UIUC/is310-fall-2024/discussions/4#discussioncomment-10695817) but just wanted to flag that you did not include your **Hypothesis** or **Discord** handles in your `README.md`. Please add those when you get a moment. Thanks 👍

**Reply by yunchaecho2** [reply]
I've just updated initial homework based on your feedback! Thank you Professor LeBlanc 👍 
https://github.com/yunchaecho2/is310-coding-assignments/tree/main/init-is310-homework

**Comment by henRotunno** [comment]
Henry Rotunno 
https://github.com/henRotunno/is310-coding-assignments

**Reply by ZoeLeBlanc** [reply]
Hi @henRotunno ! This version is very close but noticed a few issues for you to fix.

1. First, you can actually just embed your images through using the following Markdown syntax `![Alt Text](path to image)`. So in your case, your git version for example would be `![Git Version](images/git-ver.png)`.
2. Second, you have an `.Rhistory` file in your folder. This is a private environment file and so we try not to add this to our GitHub repositories since it can cause git merge errors over time.

Here's the steps to remove `.Rhistory` file:

- First, I would add `.Rhistory` to your `.gitignore` under `.DS_Store`
- Then, I would delete the `.Rhistory` file locally (so likely `rm -rf .Rhistory` from your `init-is310-homework` folder).
- Then I would do our normal git workflow of `git add.`, then `git commit -m "deleted Rhistory"`, and finally `git push origin main`

3. Third and final is that you have two additional files `havefun` and `first_page.html` in the folder. If you could also remove those and then do the same git workflow that would be great!

Great work overall 👍

**Comment by Rebe-Rod** [comment]
Updated link!

https://github.com/Rebe-Rod/is310-coding-assingments/tree/main/init-is310-homework

**Comment by elee33** [comment]
https://github.com/elee33/is310-coding-assignments/blob/main/README.md

**Reply by ZoeLeBlanc** [reply]
Hi @elee33 ! I don't see any text in your `README.md` or an `init-is310-homework` folder. Please ping me here when you add those and happy to answer questions if you are having problems 👍

**Reply by elee33** [reply]
https://github.com/elee33/is310-coding-assignments/blob/main/README.md

**Reply by ZoeLeBlanc** [reply]
Hi @elee33 ! Thanks for posting this. A few issues I noticed. First your `README.md` is not in the `init-is310-homework` folder. Second, it looks like some of the images aren't rendering properly due to the file paths not being correct. Finally, I noticed you have two `.DS_Store` files and no `.gitignore`. I'll mark this as complete but future assignments you will lose points for these issues. Feel free to reach out if you have questions about how to fix these problems!
